python/pfs/pipe2d/generateCommands.py

```diff
@@ -0,0 +1,1493 @@
+
+        Returns
+        -------
+        sourceFilter : `SourceFilter`
+        """
+        id = yamlBlock.get(key, [])
+        if isinstance(id, str):
+            id = [id]
+        if not isinstance(id, list):
+            raise RuntimeError(f"'{key}' must be a string or a list of string: '{id}'")
+
+        return cls(ensureKeyEqValue(str(x)) for x in id)
+
+    def getCommandLine(self, *, key: str = "id") -> List[str]:
+        """Get command line options for this source filter.
+
+        Parameters
+        ----------
+        key : `str`
+            If you give this argument, ``--{key}`` will be used
+            instead of ``--id`` for the command line option.
+
+        Returns
+        -------
+        options : `List[str]`
+            Like ["--id", "field=BIAS"]
+        """
+        if self.id:
+            commands = [f"--{key}"] + self.id
+        else:
+            commands = []
+        return commands
+
+
+@export
+class InitSource:
+    """Initial calibs that are necessary in order to start creating calibs.
+
+    Parameters
+    ----------
+    dirName : `str`
+        Path to a directory where files exist.
+    detectorMapFmt : `str`
+        Name of detectorMap files.
+        Occurrences of "{arm}" in it will be replaced by "r1", "b1" etc.
+    arms : `Iterable[str]`
+        ["r1", "b1"] for example.
+    """
+
+    __slots__ = ["dirName", "detectorMapFmt", "arms"]
+
+    def __init__(self, dirName: str, detectorMapFmt: str, arms: Iterable[str]):
+        self.dirName = dirName
+        self.detectorMapFmt = detectorMapFmt
+        self.arms = list(arms)
+
+    @classmethod
+    def fromYaml(cls, yamlBlock: Mapping[str, Any]) -> "InitSource":
+        """Construct ``InitSource`` from a YAML block.
+
+        Parameters
+        ----------
+        yamlBlock : `Mapping[str, Any]`
+            A block of a YAML data structure:
+
+            ``"dirName"``
+                ``"$DRP_STELLA_DATA_DIR/raw"`` for example.
+            ``"detectorMapFmt"``
+                ``"detectorMap-sim-{arm}.fits"`` for example.
+                "{arm}" in it will be replaced by "r1", "b1" etc.
+            ``"arms"``
+                ``["r1", "b1"]`` for example.
+
+        Returns
+        -------
+        initSource : `InitSource`
+        """
+        return cls(yamlBlock["dirName"], yamlBlock["detectorMapFmt"], yamlBlock["arms"])
+
+    def execute(self, logger: lsst.log.Log, fout: TextIO, dataDir: str, calib: str):
+        """Put to ``fout`` commands to ingest this source.
+
+        Parameters
+        ----------
+        logger : `lsst.log.Log`
+            Logger.
+        fout : `TextIO`
+            Output file where to write the command.
+        dataDir : `str`
+            Root of data repository.
+        calib : `str`
+            Name of output calibration directory.
+        """
+        initDir = os.path.join(dataDir, os.path.expandvars(self.dirName))
+        logger.info(f"Reading init files from {initDir}")
```

Comment:
When using the LSST logger, you should use the old `%` formatting, and include the variables as parameters, rather than interpolating them. This allows the code to only do the string manipulation if it's warranted by the log level. (Not too important here, but a good habit to get into.)

    logger.info("Reading init files from %s", initDir)

Reply:
I have modified `logger` usages.